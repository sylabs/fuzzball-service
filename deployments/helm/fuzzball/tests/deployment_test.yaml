suite: Deployment
templates:
- deployment.yaml
tests:
- it: should render by default
  asserts:
  - matchSnapshot: {}
- it: "should have name 'RELEASE-NAME-NameOverride'"
  set:
    nameOverride: NameOverride
  asserts:
  - matchSnapshot: {}
- it: "should have name 'NameOverride'"
  set:
    fullnameOverride: NameOverride
  asserts:
  - matchSnapshot: {}
- it: "should have serviceAccount 'default'"
  set:
    serviceAccount.create: false
  asserts:
  - matchSnapshot: {}
- it: "should have serviceAccount 'SvcAcctName'"
  set:
    serviceAccount.name: SvcAcctName
  asserts:
  - matchSnapshot: {}
- it: "should have 10 replicas"
  set:
    replicaCount: 10
  asserts:
  - matchSnapshot: {}
- it: "should have imagePullSecrets"
  set:
    image:
      pullSecrets:
      - name: regcred
  asserts:
  - matchSnapshot: {}
- it: "should have podSecurityContext"
  set:
    podSecurityContext:
      fsGroup: 2000
  asserts:
  - matchSnapshot: {}
- it: "should have securityContext"
  set:
    securityContext:
      readOnlyRootFilesystem: true  
      runAsNonRoot: true
  asserts:
  - matchSnapshot: {}
- it: "should have image 'repo'"
  set:
    image:
      repository: repo
  asserts:
  - matchSnapshot: {}
- it: "should have image tag 'TagOverride'"
  set:
    image:
      tag: "TagOverride"
  asserts:
  - matchSnapshot: {}
- it: "should have imagePullPolicy 'Always'"
  set:
    image:
      pullPolicy: Always
  asserts:
  - matchSnapshot: {}
- it: "should have resource limits"
  set:
    resources:
      limits:
        cpu: 100m
  asserts:
  - matchSnapshot: {}
- it: "should have nodeSelector"
  set:
    nodeSelector:
      disktype: ssd
  asserts:
  - matchSnapshot: {}
- it: "should have tolerations"
  set:
    tolerations:
    - key: "example-key"
      operator: "Exists"
      effect: "NoSchedule"
  asserts:
  - matchSnapshot: {}
- it: "should have affinity"
  set:
    affinity:
      podAntiAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
        - labelSelector:
            matchExpressions:
            - key: app
              operator: In
              values:
              - store
          topologyKey: "kubernetes.io/hostname"
  asserts:
  - matchSnapshot: {}
- it: "should have OAuth2 audience 'AudienceOverride'"
  set:
    oAuth2:
      audience: AudienceOverride
  asserts:
  - matchSnapshot: {}
- it: "should have OAuth2 issuer 'IssuerOverride'"
  set:
    oAuth2:
      issuerURI: IssuerOverride
  asserts:
  - matchSnapshot: {}
