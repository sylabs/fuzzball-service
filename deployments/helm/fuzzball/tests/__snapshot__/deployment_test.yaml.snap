should have 10 replicas:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 10
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have OAuth2 audience 'AudienceOverride':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: OAUTH2_AUDIENCE
              value: AudienceOverride
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have OAuth2 issuer 'IssuerOverride':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: OAUTH2_ISSUER_URI
              value: IssuerOverride
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have affinity:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
              - labelSelector:
                  matchExpressions:
                  - key: app
                    operator: In
                    values:
                    - store
                topologyKey: kubernetes.io/hostname
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have image 'repo':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: repo:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have image tag 'TagOverride':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:TagOverride
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have imagePullPolicy 'Always':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: Always
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have imagePullSecrets:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          imagePullSecrets:
          - name: regcred
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have name 'NameOverride':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: NameOverride
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: NameOverride
should have name 'RELEASE-NAME-NameOverride':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: NameOverride
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-NameOverride
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: NameOverride
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: NameOverride
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-NameOverride
should have nodeSelector:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          nodeSelector:
            disktype: ssd
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have podSecurityContext:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext:
            fsGroup: 2000
          serviceAccountName: RELEASE-NAME-fuzzball
should have resource limits:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources:
              limits:
                cpu: 100m
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have securityContext:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext:
              readOnlyRootFilesystem: true
              runAsNonRoot: true
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
should have serviceAccount 'SvcAcctName':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: SvcAcctName
should have serviceAccount 'default':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: default
should have tolerations:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
          tolerations:
          - effect: NoSchedule
            key: example-key
            operator: Exists
should render by default:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/name: fuzzball
        app.kubernetes.io/version: 0.2.0
        helm.sh/chart: fuzzball-0.4.0
      name: RELEASE-NAME-fuzzball
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: fuzzball
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: fuzzball
        spec:
          containers:
          - env:
            - name: MONGO_URI
              value: mongodb://server:changeme@RELEASE-NAME-mongodb/server
            - name: NATS_URIS
              value: nats://server:changeme@RELEASE-NAME-nats-client
            - name: REDIS_URI
              value: redis://:changeme@RELEASE-NAME-redis-master
            image: server:0.2.0
            imagePullPolicy: IfNotPresent
            name: fuzzball
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-fuzzball
